src/App.jsx

```diff
@@ -1,24 +1,137 @@
+import { ChromePicker } from "react-color";
+import { useEffect, useRef } from "react";
+import image2 from "../src/assets/image2.jpg";
+import "animate.css";
+import Lottie from "lottie-react";
+import animationData from "./lottie/Animation - 1699365461515.json";
 
-// Define the App component
 export const App = () => {
-  // Destructure variables, properties and methods from the useQRCodeGenerator hook that you imported above here :)
-
-  // Return the JSX to render the component
+  const {
+    url,
+    setUrl,
+    qrCode,
+    visibility,
+    generateQRCode,
+    downloadQRCode,
+    repeatAction,
+    qrCodeColor,
+    setQRCodeColor,
+    size,
+    setSize,
+    displayColorPicker,
+    setDisplayColorPicker,
+    isUrlEmpty,
+  } = useQRCodeGenerator();
+  // const lottieRef = useRef();
+  const inputRef = useRef(null);
+  useEffect(() => {
+    inputRef.current.focus();
+  }, []);
+  // Check if the selected size is "250" or "350" to show the Lottie animation
+  const isLottieVisible = size === "250" || size === "320";
   return (
-    <div className="">
+    <main className="text-white flex flex-col items-center font-poppins max-w-lg h-auto mx-auto my-10 px-4 py-10 rounded-3xl bg-primary lg:my-10 md:my-[5rem] animate__animated animate__fadeInLeft ">
       {/* Render the title */}
-      <img className="logo" src={logo} alt="" />
-      <h1>Technigo QR Code Generator</h1>
-      <p>Start Here</p>
-      <QrExample />
+      <h1 className="text-xl font-bold animate__animated  animate__zoomIn lg:text-3xl md:text-3xl">
+        Create a personal QR-Code
+      </h1>
+
+      <p className="text-yellow ">by Huadan</p>
+
+      {/* Conditionally render based on whether the user is inputting a URL or viewing the QR code */}
+
+      <div
+        className={`flex flex-col items-center ${
+          visibility ? "block" : "hidden"
+        }`}
+      >
+        <img
+          src={image2}
+          alt="image2"
+          className="rounded-[30px] w-full h-[30vh] justify-self-end my-6 lg:w-full md:h-[30vh]"
+        />
+        <input
+          ref={inputRef}
+          type="text"
+          placeholder="e.g https://google.com"
+          value={url}
+          onChange={(e) => setUrl(e.target.value)}
+          className="w-[80%] h-[3rem] rounded-full mb-6 text-primary placeholder:italic placeholder:pl-6 focus:outline-none focus:border-[3px] focus:border-green focus:ring-1 focus:ring-green block  px-5 py-2 lg:w-full "
+        />
+
+        <button
+          onClick={() => setDisplayColorPicker(!displayColorPicker)}
+          className="bg-green text-primary w-[80%] lg:w-[25rem] h-12 rounded-lg font-bold cursor-pointer "
+        >
+          Pick Color
+        </button>
+        {displayColorPicker ? (
+          <div>
+            <ChromePicker
+              color={qrCodeColor}
+              onChange={(color) => setQRCodeColor(color.hex)}
+            />
+          </div>
+        ) : null}
```

Comment:
Consider using input with a type of color: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/color
This would solve all your needs without having to add a dependency to another package.